### PR TITLE
CARGONIA THE FREE: The Quartermaster is now a head of staff.

### PR DESCRIPTION
**Pull request by Iamgoofball**
## About The Pull Request
dreamseeker_A16fbkAdwL

## Why It's Good For The Game

oranges has stated multiple times that the quartermaster is a head of staff, that he intends to make the quartermaster a head of staff, and that the QM should be treated as a head of staff. As such, I've made it official with this PR.

source, if anyone doubts me that oranges wants the QM to be a head of staff:
Discord_xWFDIXXLfV
Discord_dGwvPk9Jbr

this is also a precursor to us shutting off antag for heads of staff, potentially

QM as a head of staff is way more fun to play. You can coordinate easier with other departments via :C, which comes in incredibly handy during high volume situations where people are ignoring common radio.

Having a telebaton and ID access means the QM can now actually demote his subordinates. If the cargo tech or miner mouths off and refuses to do the job, he can actually demote their ass without having to hope the HoP is in his office(he never is), remembers he runs cargo(he never does), and knows how to demote someone(surprisingly, a lot of HoPs don't know what to do in this situation from my experience).

QM as a fake head of staff made sense back in 2012 when cargo was "move crates and wait for the miner to get back with the shift's ore".

Nowadays, cargo has bounties, drones, the entire economy system, ore management, a massive mining rework, mail, and all kinds of other minor stuff that makes the department bigger than it was before. It's enough to have it's own head of staff, and Cargo can finally escape the faux-department purgatory where it was forced to share a head of staff with another department.

## Changelog
:cl: Iamgoofball
balance: The Quartermaster is now a Head of Staff, and answers directly to the Captain now.
balance: This comes with all the stuff a Head of Staff normally has, like command access, a telebaton, and a silver ID.
balance: This also comes with no longer being eligible for Head Revolutionary, and being a target for the Revolution.
balance: Thanks to a savvy contract with the Space Teamsters, non-humans are allowed to be Quartermasters still.
balance: The HoP is no longer the lead of Supply, nor does he have access to Supply.
balance: The Warden now carries the torch for pretender royal metabolism, as the last remaining pseudo-head.
code: Removes a hack from the NT IRN code.
/:cl:

### CODE REVIEW

**Comment by ANAME593** [comment]
Finally a goof pr that isnt shit

File: code/modules/modular_computers/computers/item/role_tablet_presets.dm
Comment (JohnFulpWillard) [inline code/modules/modular_computers/computers/item/role_tablet_presets.dm:235]: move this to command section, at the very top

**Review by JohnFulpWillard — CHANGES_REQUESTED** [review]
budget orders doesnt need the snowflake QM check anymore

**Review by JohnFulpWillard — CHANGES_REQUESTED** [review]
QM access has to be moved to PRIVATE_COMMAND_ACCESS*, also you have to actually map in the QM's new command equipment in their office (and probably, reinforced windows in meta).

**Comment by Shadowflame909** [comment]
Should just merge them with the hop

**Comment by Fikou** [comment]
https://github.com/tgstation/tgstation/blob/adcb8079452a6ee78dc20ae8393ec2bf1ad62708/strings/sillytips.txt#L17

**Comment by bluetshirtguy** [comment]
The QM not being able to be a Xeno is a tragedy

**Comment by distributivgesetz** [comment]
curious about how this is going to affect revolutions

**Comment by ANAME593** [comment]
> curious about how this is going to affect revolutions

for starters revs will need to kill one extra head to win and QM is usually very strong as a rev cuz he can arm his fellow revs easily now the revs will have to kill the QM and take his ID to arm themselves and also stop security from being able to access cargo

**Comment by Ryusenshu** [comment]
Thanks
i always wanted one less job i wanna play

File: code/game/gamemodes/dynamic/dynamic_rulesets_latejoin.dm
Comment (Mothblocks) [inline code/game/gamemodes/dynamic/dynamic_rulesets_latejoin.dm:96]: abc order

File: code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm
Comment (Mothblocks) [inline code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm:483]: abc order

**Comment by Mothblocks** [comment]
@optimumtact urs

**Comment by GuillaumePrata** [comment]
> balance: Thanks to a savvy contract with the Space Teamsters, non-humans are allowed to be Quartermasters still.

A first step at last!

**Comment by distributivgesetz** [comment]
Also, since QMs have communication console access now, they should be able to purchase any shuttle they want (since it's their money)

**Comment by Megarop** [comment]
I don't get the reasoning behind letting the QM be an alien. If you don't like human-only heads then remove it, it's just weird if all the other head roles are human only but then one of them isn't for no apparent reason.

**Comment by Snowyypanda** [comment]
> If you don't like the human-only heads rule then just remove it, it's just weird if all the other head roles are human only but then one of them isn't for no apparent reason.

true, either make all head roles species uniform or just don't make qm a head role

**Comment by SoreYew** [comment]
First non-human Head of Staff, we did it, fellow freaks! Q

**Comment by Comxy** [comment]
Chaplain head of church when?

**Comment by Shadowflame909** [comment]
> First non-human Head of Staff, we did it, fellow freaks!

wtf central command ending space racism?

This kills ingame lore built up over decades on the wiki pages and my immersion!!

This will also probably kill the PR if any maintainer dared take it seriously

**Comment by SoreYew** [comment]
> > First non-human Head of Staff, we did it, fellow freaks!
> 
> 
> 
> wtf central command ending space racism?
> 
> 
> 
> This kills ingame lore built up over decades on the wiki pages and my immersion!!
> 
> 
> 
> This will also probably kill the PR if any maintainer dared take it seriously

If you read the pr its clearly mentioned this is a labor win from the Space Teamsters (Spinward local 14294 represent!) so idk what you are on about.

**Comment by wesoda25** [comment]
>making the QM a head of staff improves the QM experience, [SNIP] , and really makes cargo more fun. 

Could you explain why?

**Comment by Fikou** [comment]
> making the QM a head of staff improves the QM experience, [SNIP] , and really makes cargo more fun.

explain this in non vague terms please. i think that the dynamic of a side of the station that isnt as much aligned with command/security is cool as fuck, as well as the revs dynamic with the qm being a hard but really good conversion target, with cargo being an important place for both sec and revs

also making qm a head and then slapping a shit snowflake "this head can be nonhuman" is dumb as fuck

**Comment by Iamgoofball** [comment]
It was done at headcoder orders so take it up with mothblocks

**Comment by wesoda25** [comment]
Didn't we _just_ have an update to contrib guidelines stating that you have you actually justify your changes and not just point at a maint who wants it?

**Comment by Iamgoofball** [comment]
i didn't point at a maintainer who wants it, i pointed at the headcoder wanting it

**Comment by normie5-cell** [comment]
God no... They have HoP already, what's the point?!

**Comment by Iamgoofball** [comment]
99 out of 100 hops remember they're the boss of cargo

**Comment by Fikou** [comment]
> i didn't point at a maintainer who wants it, i pointed at the headcoder wanting it

https://github.com/tgstation/tgstation/blob/345d1e71d5d8bf25bec627ac55ba769186326624/.github/CONTRIBUTING.md?plain=1#L150-L152

**Comment by SuperNovaa41** [comment]
image

**Comment by NamelessFairy** [comment]
> 99 out of 100 hops never remember they're the boss of cargo

I assume you have round data to back this up. Mind posting it?

**Comment by SuperNovaa41** [comment]
Come back when you aren't in the negative GBP

**Comment by SuperNovaa41** [comment]
Welcome back

File: code/__DEFINES/access.dm
Comment (Son-of-Space) [inline code/__DEFINES/access.dm:313]: please abc order this too, see #67512 .

**Comment by ATH1909** [comment]
> balance: The HoP is no longer the lead of Supply, nor does he have access to Supply.

Don't think I didn't notice you trying to sneak that in there.

**Comment by datorangebottle** [comment]
I like everything about this except the nonhuman QM snowflake bit. Doesn't make any sense whatsoever to have one head able to be nonhuman when every other head of staff(and the captain) are human only. Either do all of them, or none of them.

**Comment by Iamgoofball** [comment]
@ATH1909 it's literally a normal part of making the QM a head

**Comment by Dorsisdwarf** [comment]
The HoP being kicked out of supply is a good change, now maybe he'll remember he's the head of service not Mr Guns

**Comment by itseasytosee** [comment]
image
🤨

File: code/modules/jobs/job_types/_job.dm
Comment (Watermelon914) [inline code/modules/jobs/job_types/_job.dm:446]: If QM is being a head, they shouldn't get some special exemption to being non-human

**Review by Watermelon914 — CHANGES_REQUESTED** [review]

File: code/modules/jobs/job_types/_job.dm
Comment (Mothblocks) [reply code/modules/jobs/job_types/_job.dm:446]: i told goof they should because i like lizard qms

Comment (Watermelon914) [reply code/modules/jobs/job_types/_job.dm:446]: waiting on oranges response to this pr

Comment (Son-of-Space) [reply code/modules/jobs/job_types/_job.dm:446]: The deep lore spinsters union is great stuff though.

**Comment by optimumtact** [comment]
I approve this PR, also, it is funny that you can be a non human head as the QM.

**Comment by Redrover1760** [comment]
Nonhuman heads are fun. The primary reason they aren't a thing is due to the whole lore stuff around it, and balancing in revs/AI etc. Having a potential nonhuman head that's literally last in command sounds pretty okay under lore anyways, and probably opens up some inner class conflict perhaps of "real heads vs QM"? Who knows.

Not to mention removing nonhumans is just as controversial as making nonhumans heads, if not more so, so lol.

**Comment by Shadowflame909** [comment]
imagine if revs could convert 1 head of staff to their side, with a slow minute timer...

Going for the captain would be powerful...but the QM practical.

**Comment by wesoda25** [comment]
I'm curious if some responsibility might be projected onto cargo from this. HOP paying more attention to service will also be nice to see.

**Comment by optimumtact** [comment]
you need to make the non human QM bit a config defaulting to off.

**Comment by MrStonedOne** [comment]
Bobdobbington told me to approve this pr but i can't approve goofball bypassing configs so goof your gonna have to add a new config for enforcing human authority on QMs.

Also I strongly encourage anybody playing hop or captain or really any head of staff, to treat QM as Not A Real Head Role:tm: on any server that doesn't enforce human authority.

**Comment by MrStonedOne** [comment]
> you need to make the non human QM bit a config defaulting to off.

oranges is a dumbass and can be ignored goof.

the head of staff non-human config is setup such that all races are allowed by default, and you turn a config flag on to only allow humans.

so the QM flag should work the same way.

**Comment by Iamgoofball** [comment]
this isn't a config though this is a new addition

File: code/modules/antagonists/traitor/objectives/bug_room.dm
Comment (JohnFulpWillard) [inline code/modules/antagonists/traitor/objectives/bug_room.dm:28]: trailing comma here please

File: code/modules/modular_computers/computers/item/role_tablet_presets.dm
Comment (JohnFulpWillard) [inline code/modules/modular_computers/computers/item/role_tablet_presets.dm:125]:
```suggestion
/obj/item/modular_computer/tablet/pda/heads/quartermaster/Initialize(mapload)
```

**Comment by Wallemations** [comment]
Does this make the HOP only the head of the service department?

**Comment by optimumtact** [comment]
Fix your lint issues

File: code/datums/id_trim/jobs.dm
Comment (JohnFulpWillard) [inline code/datums/id_trim/jobs.dm:201]: throughout this file please fix the merge problems

File: code/game/machinery/PDApainter.dm
Comment (JohnFulpWillard) [inline code/game/machinery/PDApainter.dm:381]:
```suggestion
/// Supply departmental variant. Limited to PDAs defined in the SSid_access.sub_department_managers_tgui data structure.
```

**Comment by castawaynont** [comment]
massive L

**Pull request merged**
